src/instance.ts

```diff
@@ -16,13 +16,101 @@
 
 import arrify = require('arrify');
 const common = require('@google-cloud/common-grpc');
-import {ServiceObjectConfig} from '@google-cloud/common';
+import {ServiceObjectConfig, GetConfig} from '@google-cloud/common';
 import {paginator} from '@google-cloud/paginator';
 import {promisifyAll} from '@google-cloud/promisify';
 import * as extend from 'extend';
-import * as is from 'is';
 import snakeCase = require('lodash.snakecase');
 import {Database} from './database';
+import {google as instanceAdmin} from '../proto/spanner_instance_admin';
+import {google as databaseAdmin} from '../proto/spanner_database_admin';
+import {Operation as GaxOperation} from 'google-gax';
+import {
+  SessionPoolOptions,
+  SessionPoolInterface,
+  SessionPool,
+} from './session-pool';
+import {Spanner} from '.';
+import {ServiceError} from 'grpc';
+import {RequestCallback, PagedResponse} from './common';
+import {Stream} from 'stream';
+
+export type IDatabase = databaseAdmin.spanner.admin.database.v1.IDatabase;
+export type IInstance = instanceAdmin.spanner.admin.instance.v1.IInstance;
+export type ILongrunningOperation = instanceAdmin.longrunning.IOperation;
+export type CreateDatabaseResponse = [
+  Database,
+  GaxOperation,
+  ILongrunningOperation
+];
+export type DeleteInstanceResponse = [instanceAdmin.protobuf.IEmpty];
+export type ExistsInstanceResponse = [boolean];
+export type GetInstanceResponse = [Instance, IInstance];
+export type GetMetadataResponse = [IInstance];
+export type GetDatabasesResponse = PagedResponse<
+  Database,
+  databaseAdmin.spanner.admin.database.v1.IListDatabasesResponse
+>;
+export type SetMetadataResponse = [GaxOperation, ILongrunningOperation];
+
+export interface CreateDatabaseOptions
+  extends databaseAdmin.spanner.admin.database.v1.ICreateDatabaseRequest {
+  poolOptions?: SessionPoolOptions;
+  poolCtor?: SessionPool;
+  schema?: string;
+}
+export interface CreateInstanceRequest extends IInstance {
+  nodes: number;
+}
+export interface GetDatabasesRequest
+  extends databaseAdmin.spanner.admin.database.v1.IListDatabasesRequest {
+  autoPaginate?: boolean;
+  maxApiCalls?: number;
+  maxResults?: number;
+}
+export interface CreateInstanceCallback {
```

Comment:
You might consider creating a generic for calls like this, `LongRunningCallback<Resource>`?